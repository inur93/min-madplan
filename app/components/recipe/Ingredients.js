import { List, ListHeader } from "semantic-ui-react";


const ListItem = ({ ingredient }) => {
    const { amount, unit, name } = ingredient;
    let fullname = amount ? `${amount} ${unit || ''} ${name}`
        : name;
    return <List.Item>
        <List.Content>
            <ListHeader>{fullname}</ListHeader>
        </List.Content>
    </List.Item>
}

export function Ingredients({ loading, recipe }) {

    return <List>
        {recipe.ingredients.map(
            i => <ListItem key={i._id} ingredient={i} />
        )}
    </List>
}