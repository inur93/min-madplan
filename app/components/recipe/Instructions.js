import Markdown from 'react-markdown';

export const Instructions = function({ source }) {
    return (
        <Markdown source={source} />
    )
}