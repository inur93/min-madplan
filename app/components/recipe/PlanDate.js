import { Message, Icon, List as ListSUI } from 'semantic-ui-react';
import { formatDay } from '../../functions/dateFunctions';


export function PlanDate({ loading, plan, date }) {
    if (loading) return null;
    const selected = plan.find(x => x.date === date);
    return <Message info
        header={formatDay(new Date(date))}
        content={selected ? selected.recipe.name : 'VÃ¦lg en opskrift...'} />
}