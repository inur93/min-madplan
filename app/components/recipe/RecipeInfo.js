import { Icon } from "semantic-ui-react";



export function RecipeInfo({ recipe }) {

    return (<div id='stats'>
        <h2>{recipe.title}</h2>
        <p>
            Denne side skal bl.a. indeholde hvor lang tid opskriften tager og evt andre nyttige information.
    I mellemtiden brug <Icon name='numbered list' /> og <Icon name='clipboard list' /> for at se fremgangsm√•de og ingredienser.
</p></div>)
}