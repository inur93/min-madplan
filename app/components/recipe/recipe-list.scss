@import '../../styles/variables.scss';
div.item.recipe-list-item .content {
    display: flex;
    flex-flow: row;
    align-items: stretch;

    font-size: 18px;
    .header {
        flex: auto 1 1;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        margin-right: 0.5rem;
    }

    .icon {
        flex: auto 0 1;
        padding: unset;
        margin: auto 0;
        padding-left: 0.5rem;

        &.check {
            color: $color-positive;
        }
        
    }
}
