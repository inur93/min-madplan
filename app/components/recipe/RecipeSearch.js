import { useRouter } from 'next/router';
import FlipMove from 'react-flip-move';
import { Button, List } from 'semantic-ui-react';
import { usePlanDay } from '../../hooks/plan/usePlanDay';
import { usePlanDetails } from '../../hooks/plan/usePlanDetails';
import { useRecipeSearch } from '../../hooks/recipe/useRecipeSearch';
import { useView } from '../../hooks/useView';
import { IconCheck, IconInfo } from '../shared/Icon';
import { SearchInput } from '../shared/Input';
import { Loader } from '../shared/Loader';
import { forwardRef } from 'react';

const RecipeListItem = forwardRef(({ recipe }, ref) => {
    const { _id, title } = recipe;
    const [state] = usePlanDay('plan', 'date');
    const router = useRouter();
    const [, planHandlers] = usePlanDetails('plan');
    const [, , goTo] = useView('/recipes');

    const onCheck = () => planHandlers.updatePlanDay({
        id: router.query.plan,
        date: state.planDay.date,
        recipe: recipe
    })
    const showInfo = () => goTo.details(_id, true);
    const onSelect = () => state.hide ? showInfo() : onCheck()
    return (<List.Item ref={ref} >
        <List.Content>
            <List.Header onClick={onSelect}>{title}</List.Header>
        </List.Content>
        <List.Content>
            {!state.hide && <IconInfo onClick={showInfo} />}
            {!state.hide && <IconCheck onClick={onCheck} />}
        </List.Content>
    </List.Item>)
});

export function RecipeSearch() {
    const [state, handlers] = useRecipeSearch();
    return (<div>
        <SearchInput loading={state.loading} placeholder="Søg efter opskrift..." />
        <p>{state.results.length} ud af {state.count}</p>
        <Loader loading={state.loading}>
            <List className='recipe-list' selection verticalAlign='middle'>
                <FlipMove>
                    {state.results.map(r => <RecipeListItem key={r._id} recipe={r} />)}
                </FlipMove>
            </List>
            <Button disabled={state.isLoadingMore || state.isReachingEnd}
                loading={state.isLoadingMore}
                fluid
                onClick={handlers.loadMore}>
                {state.isReachingEnd ? 'der er ikke flere resultater' : 'indlæs flere'}
            </Button>
        </Loader>
    </div>)
}