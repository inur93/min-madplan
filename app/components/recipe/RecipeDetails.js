import React from 'react';
import { useRecipeDetails } from "../../hooks/recipe/useRecipeDetails";
import { Loader } from "../shared/Loader";
import { Page } from "../shared/Page";
import { ScrollView } from "../shared/ScrollView";
import { Ingredients } from "./Ingredients";
import { Instructions } from "./Instructions";
import { RecipeInfo } from "./RecipeInfo";


export function RecipeDetails() {
    const [state] = useRecipeDetails();
    
    return <Loader loading={state.loading}>
        <Page>
            <ScrollView>
                <RecipeInfo recipe={state.recipe} />
                <Ingredients recipe={state.recipe} />
                <Instructions recipe={state.recipe} />
            </ScrollView>
        </Page>
    </Loader>
}