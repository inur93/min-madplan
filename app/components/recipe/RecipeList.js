import './recipe-list.scss';
import { Message, Icon, List } from 'semantic-ui-react';
import { IconInfoCircle, IconInfo, IconCheck } from '../shared/Icon';

function RecipeListItem({ id, title, onClick, onShowDetails, showActions }) {
    const onSelect = () => onClick({ id });
    const showDetails = () => onShowDetails({ id });
    return (<List.Item className='recipe-list-item'>
        <List.Content>
            <List.Header onClick={onSelect}>{title}</List.Header>
        </List.Content>
        <List.Content>
            {showActions && <IconInfo onClick={showDetails} />}
            {showActions && <IconCheck onClick={onSelect} />}
        </List.Content>
    </List.Item>)
}

export function RecipeList({ recipes, onClick, onShowDetails, showActions }) {

    return (<List className='recipe-list' selection verticalAlign='middle'>
        {recipes.map(recipe => <RecipeListItem key={recipe._id}
            id={recipe._id}
            title={recipe.title}
            onClick={onClick}
            showActions={showActions}
            onShowDetails={onShowDetails} />)}
    </List>)
}