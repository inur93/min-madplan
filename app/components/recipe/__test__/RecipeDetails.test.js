import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { RecipeDetails } from "../RecipeDetails";
import renderer from "react-test-renderer";
import { useRecipeDetails } from '../../../hooks/recipe/useRecipeDetails';
jest.mock("../../../hooks/recipe/useRecipeDetails");

let container = null;
beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
})

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
})

it("render headers, ingredients and instructions", () => {
    useRecipeDetails.mockReturnValue([{
        loading: false,
        recipe: {
            title: 'f√∏rste test opskrift',
            ingredients: [
                { _id: '1', name: 'kartofler', unit: 'kg', amount: 1 },
                { _id: '2', name: 'sukker', unit: 'g', amount: 2 }
            ],
            instructions: "En beskrivelse af hvordan man koger kartofler."
        }
    }])
    const tree = renderer
        .create(<RecipeDetails />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})


it("render loading while waiting for recipe", () => {
    useRecipeDetails.mockReturnValue([{
        loading: true
    }])

    const tree = renderer
        .create(<RecipeDetails />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})