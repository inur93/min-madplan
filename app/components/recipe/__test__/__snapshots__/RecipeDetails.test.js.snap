// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render headers, ingredients and instructions 1`] = `
<div
  className="cover-height"
>
  <div
    className="page-component"
  >
    <div
      className="flex row"
    >
      <div
        className="ui small inline loader shrink"
      />
    </div>
    <div
      className="scroll-view"
    >
      <div
        id="stats"
      >
        <h2>
          første test opskrift
        </h2>
        <p>
          Denne side skal bl.a. indeholde hvor lang tid opskriften tager og evt andre nyttige information. I mellemtiden brug 
          <i
            aria-hidden="true"
            className="numbered list icon"
            onClick={[Function]}
          />
           og 
          <i
            aria-hidden="true"
            className="clipboard list icon"
            onClick={[Function]}
          />
           for at se fremgangsmåde og ingredienser.
        </p>
      </div>
      <div
        id="ingredients"
      >
        <h2>
          Dette skal du bruge
        </h2>
        <div
          className="ui list"
          role="list"
        >
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <div
              className="content"
            >
              <div
                className="header"
              >
                1 kg kartofler
              </div>
            </div>
          </div>
          <div
            className="item"
            onClick={[Function]}
            role="listitem"
          >
            <div
              className="content"
            >
              <div
                className="header"
              >
                2 g sukker
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        id="instructions"
      >
        <h2>
          Sådan gør du
        </h2>
        <p>
          En beskrivelse af hvordan man koger kartofler.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`render loading while waiting for recipe 1`] = `
<div
  className="ui massive active text loader"
>
  loading...
</div>
`;
